src/helpers/graph.js

```diff
@@ -1,38 +1,45 @@
-export function removeNoteFromGraph(nodes, edges, noteToDelete) {
+function doesDeletedNodeHasSiblings(edges, parentId) {
+  return edges.find(edge => edge.from === parentId);
+}
+
+function revokeParentStatus(nodes, parentId) {
+  return nodes.map(node => {
+    return node.id === parentId ? { ...node, group: 'children' } : node;
+  });
+}
+
+const hasChildren = (edges, nodeId) =>
+  Boolean(edges.find(edge => edge.from === nodeId));
+
+export function removeNodeFromGraph(nodes, edges, nodeToDelete) {
   let newNodes = [...nodes];
   let newEdges = [...edges];
-  let parentEdge = newEdges.find(edge => edge.to === noteToDelete.id);
+  let parentId = newEdges.find(edge => edge.to === nodeToDelete.id).from;
 
-  removeChild(noteToDelete.id);
-  newNodes = newEdges.find(edge => edge.from === parentEdge.from) //check if there are siblings of deleted node if not revoke parant status
+  removenode(nodeToDelete.id);
+
+  newNodes = doesDeletedNodeHasSiblings(newEdges, parentId)
     ? newNodes
-    : newNodes.map(node => {
-        return node.id === parentEdge.from
-          ? { ...node, group: 'children' }
-          : node;
-      });
-  return { notes: newNodes, edges: newEdges };
+    : revokeParentStatus(newNodes, parentId);
+  return { nodes: newNodes, edges: newEdges };
 
-  function removeChild(childNoteId) {
-    newNodes = newNodes.filter(note => note.id !== childNoteId);
-    if (newEdges.find(edge => edge.from === childNoteId)) {
-      // check if the node has children
+  function removenode(nodeId) {

```

Comment:
1. removeNode (camel case)
2. We have a function named `removeNodeFromGraph` and a function named `removenode`. What is the difference between the two? Does `removenode` removes node not from graph but from somewhere else? I believe it's confusing.

Reply:
Done